src/main/java/com/mixpanel/android/viewcrawler/EditProtocol.java

```diff
@@ -52,9 +52,14 @@ public CantGetEditAssetsException(String message, Throwable cause) {
         }
     }
 
-    public EditProtocol(ResourceIds resourceIds, ImageStore imageStore) {
+    public interface OnErrorListener {
```

Comment:
I don't think we can do that. This is because EditProtocol is initialized in ViewCrawler and EditProtocol initializes Visitor. And because of that, to be able to pass OnErrorListener to Visitor, we need to pass an instance of ViewCrawler to EditProtocol.